### ISSUE TITLE
Build the precise engine and disable rpath or dlopen

### ISSUE THREAD

**Issue by livingsilver94** [MycroftAI/mycroft-precise#106]
Hello,
I'm trying to improve the Mycroft package for Solus. For this purpose it would be cool to be able to build the precise-engine binary against our libs and disable its rpath/dlopen behavior, so that it's easier to keep track of its dependencies and, even better, control how and at which version these deps are. For example, we could solve the Tensorflow AVX issue by our own, or make precise-engine use our Python instead of the one that is downloaded in the precise-engine tarball.
Is this feasible?

EDIT: I said "we" to keep my message simple, but I must make clear that I'm a voluntary Solus contributor and I do not represent any member of the Solus development team.

**Comment by MatthewScholefield** [comment]
Yep, precise-engine is just a regularly packaged python script, so it could be potentially distributed like any other python binary, requiring a system installed version of tensorflow. The only concern is that if the system tensorflow version is different that the binary version, then this could cause an incompatibility with trained models. I'll leave this up to you as to whether it still makes sense (it would still always work training and using models independent of Mycroft/precise binary releases).

**Comment by livingsilver94** [comment]
Yes I know that a Python package ca be easily distributed, but I'm more talking about all the binaries like `precise-engine` and all the libraries in its directory. As far I understood, the're no way to compile these binaries on own our, so we have to rely on a prebuilt ELF file that looks for .so files in its `$PWD`. That requires us a bit of patching that would be absolutely awesome to avoid :)
